server/functions/src/index.ts

```diff
@@ -0,0 +1,126 @@
+import * as admin from "firebase-admin";
+import * as functions from "firebase-functions";
+import * as fs from "fs";
+
+admin.initializeApp();
+const db = admin.firestore();
+
+const COUNTRY_CODE = /^[A-Z][A-Z]$/; // Literal regex for immediate compilation.
+
+// FCM documentation doesn"t specify standard but usually up to 162 chars
+// https://stackoverflow.com/a/12502351/1509221
+const FCM_TOKEN_MAX_LENGTH = 4096;
+
+enum Platform {
+  IOS = 0,
+  ANDROID = 1,
+  WEB = 2,
+}
+
+interface Client {
+  uuid: string;
+  token: string;
+  disableNotifcations: boolean;
+  platform: Platform;
+  isoCountryCode: string;
+  subscribedTopics: string[];
+}
+
+// Implementation of the v1 API"s `getCaseStats` method.
+// TODO: replace with direct Firestore access from the client.
+export const getCaseStats = functions
+  .region("europe-west6")
+  .https.onRequest((request, response) => {
+    // A "fake" of the v1 API: we're just going to return a static file that we
+    // happen to have on disk.
+    // TODO: replace this with actual statistics in Firestore.
```

Comment:
for sure! given that the current thing puts stats in datastore, do you think it's easier to modify the app engine app to write to datastore-in-firestore-mode or just rewrite? would the rewrite be in typescript? It's pretty straightforward.

Reply:
I suggest we rewrite the existing App Engine code as a [scheduled Cloud Function](https://firebase.google.com/docs/functions/schedule-functions). That way we don't have to introduce AppEngine to the v2 backend at all.

Reply:
@rjhuijsman - I like the idea of using the Scheduled Cloud Function too